src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+                image_orientation=[
+                    float(v) for v in src_img.ImageOrientationPatient
+                ],
+            )
+        if plane_orientation is None:
+            plane_orientation = source_plane_orientation
+        self._plane_orientation = plane_orientation
+        self._source_plane_orientation = source_plane_orientation
+
+        # TODO: Double check correctness of DimensionIndexSequence, see
+        # discussion on Segmentation
+        # Also TODO: This includes some segmentation-specific stuff. Fix that
+        self.DimensionIndexSequence = DimensionIndexSequence(
+            coordinate_system=self._coordinate_system
+        )
+        dimension_organization = Dataset()
+        dimension_organization.DimensionOrganizationUID = (
+            self.DimensionIndexSequence[0].DimensionOrganizationUID
+        )
+        self.DimensionOrganizationSequence = [dimension_organization]
+
+        if is_multiframe:
+            self._source_plane_positions = (
+                self.DimensionIndexSequence.get_plane_positions_of_image(
+                    self._source_images[0]
+                )
+            )
+        else:
+            self._source_plane_positions = (
+                self.DimensionIndexSequence.get_plane_positions_of_series(
+                    self._source_images
+                )
+            )
+
+        shared_fg_item.PixelMeasuresSequence = pixel_measures
+        shared_fg_item.PlaneOrientationSequence = plane_orientation
+
+        # Identity Pixel Value Transformation
+        if pixel_array.dtype.kind == 'i':
+            # In case of signed integer type we rescale values to unsigned
+            # 16-bit integer range.
+            transformation_item = Dataset()
+            transformation_item.RescaleIntercept = 2 ** 16 / 2
+            transformation_item.RescaleSlope = 1
+            transformation_item.RescaleType = 'US'
+        else:
+            transformation_item = Dataset()
+            transformation_item.RescaleIntercept = 0
+            transformation_item.RescaleSlope = 1
+            transformation_item.RescaleType = 'US'
+        shared_fg_item.PixelValueTransformationSequence = [transformation_item]
+
+        # Frame VOI LUT With LUT
+        voi_lut_item = Dataset()
+        voi_lut_item.WindowCenter = window_center
+        voi_lut_item.WindowWidth = window_width
+        voi_lut_item.VOILUTFunction = 'LINEAR_EXACT'
+        shared_fg_item.FrameVOILUTSequence = [voi_lut_item]
+
+        # Parametric Map Frame Type
+        frame_type_item = Dataset()
+        frame_type_item.FrameType = self.ImageType
+        shared_fg_item.ParametricMapFrameTypeSequence = [frame_type_item]
+
+        # Real World Value Mapping Sequence
+        # If the input was a single RWVM or a sequence of size 1 then we will
+        # assign it to the Shared FG Seq. Otherwise it will be per-frame and
+        # will be checked later on
+        if len(real_world_value_mappings) == 1:
```

Comment:
Suggest a runtime check that the `real_world_value_mappings` parameter has the correct type

Reply:
Addressed via 71f8940b0a303fb5d3695c9e3068ab05f644d053

Reply:
Addressed via 71f8940b0a303fb5d3695c9e3068ab05f644d053